Source/DataSources/EntityCollection.js

```diff
@@ -30,15 +30,27 @@ define([
     };
 
     function fireChangedEvent(collection) {
+        if (collection._firing) {
```

Comment:
`_firing` is never set to true, as far as I can tell.  I think you must be missing some tests.